### PR TITLE
[torch_glow] Add support for step > 1 for slice

### PR DESCRIPTION
**Pull request by sgeorgiev255**
Summary:
This PR adds support for step>1 for the slice op.

### CODE REVIEW

**Comment by vuzelac-cadence** [comment]
@jfix71 @sgeorgiev255, I'd move this lowering into Lower or better into GraphOptimizer, if you find that moving into Lower might be too obtrusive.
Lowering in Importer, IMO, is bad. I understand it's original use (to e.g. quickly add some functionality without altering graph and propagating your changes into Glow proper) but that now goes with the need to duplicate the coding when adding support for ONNX and Pytorch.

**Comment by vuzelac-cadence** [comment]
I am realizing that the effort needed to bring in the step argument into Glow might be too big at the moment. Thus, the duplication of this lowering within ONNX, if needed, is a better approach.

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 1]
```diff
@@ -3280,9 +3280,9 @@ Error PyTorchModelLoader::loadReshape(const torch::jit::Node *ptNode) {
 
   c10::ScalarType dtype;
   RETURN_IF_ERR(getCorrectTypeMapping(dtype, inputs[ReshapeInputs::input]));
-  RETURN_ERR(addValueMapping(
-      outputs[0], F_.createReshape("reshape", input, castVector<dim_t>(shape)),
-      dtype));
+  RETURN_ERR(
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:3283]: Why change here?

File: torch_glow/src/PyTorchModelLoader.cpp [hunk 2]
```diff
@@ -5218,25 +5218,41 @@ Error PyTorchModelLoader::loadSlice(const torch::jit::Node *ptNode) {
...
+      gatherIndices.push_back(i);
     }
+    auto indicesTensor =
+        glow::Tensor(glow::ElemKind::Int64ITy, {gatherIndices.size()});
+    indicesTensor.getHandle<int64_t>() = gatherIndices;
+    auto indicesConstant = F_.getParent()->createConstant(
+        "indices_constant", std::move(indicesTensor));
+
+    output =
+        F_.createGather("gather", input, indicesConstant, dim)->getResult();
```
Comment (zrphercule) [inline torch_glow/src/PyTorchModelLoader.cpp:5250]: better name this to be sth like "sliceGeneratedGather"

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 1 above)
Comment (sgeorgiev255) [reply torch_glow/src/PyTorchModelLoader.cpp]: My bad, fixed.

File: torch_glow/src/PyTorchModelLoader.cpp (hunk 2 above)
Comment (sgeorgiev255) [reply torch_glow/src/PyTorchModelLoader.cpp]: Done.

**Pull request closed without merging**